index.js

```diff
@@ -118,6 +152,76 @@ class QuickLRU {
 
 		return Math.min(this._size + oldCacheSize, this.maxSize);
 	}
+
+	resize(newSize) {
+		if (!(newSize && newSize > 0)) {
+			throw new TypeError('`maxSize` must be a number greater than 0');
+		}
+
+		if (this._size > newSize) {
+			this._emitEvictions(this.oldCache, this.cache);
+			this._emitEvictions(slice(this.cache, 0, this._size - newSize));
+			this.oldCache = new Map(slice(this.cache, this._size - newSize));
+			this.cache = new Map();
+			this._size = 0;
+		} else if (this._size === newSize) {
+			this._emitEvictions(this.oldCache, this.cache);
+			this.oldCache = this.cache;
+			this.cache = new Map();
+			this._size = 0;
+		} else if (this._size + this.oldCache.size < newSize) {
+			this.cache = new Map(concat(this.oldCache, this.cache));
+			this.oldCache = new Map();
+			this._size = this.cache.size;
+		} else {
```

Comment:
These cases are purely for optimization, and could be omitted for brevity/readability. I'm cool with that, considering `resize` should be a rare operation.

Reply:
Yeah, I would remove omit it.